README.md

```diff
@@ -1 +1,73 @@
-wordle-dictionary-parser
+# wordle-dictionary-parser
+## Usage
+Includes usage and examples of included scripts.
+### **parse_dictionary.py**
+Parses out and separates words based on lengths and then writes them to .json files.
+#### Usage
+`usage: parse_dictionary.py [-h] -i INPUT -o OUTPUT [-fn FILENAME] [--min MIN] [--max MAX] [-t THRESHOLD] [-c]`
+```
+Outputs a .json file of words based on length and word frequency from a dictionary .json file.
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -i INPUT, --input INPUT
+                        Input file (.json dictionary). Required
+  -o OUTPUT, --output OUTPUT
+                        Output file directory. Required
+  -fn FILENAME, --filename FILENAME
+                        Output file name.
+  --min MIN             Minimum word length to be parsed into a file.
+  --max MAX             Maximum word length to be parsed into a file.
+  -t THRESHOLD, --threshold THRESHOLD
+                        Zipf frequency threshold to use if parsing out hard words.
+  -l LANG, --lang LANG  language of the word being checked (default: english 'en')
+  -c, --combine         Parse out words of same length into single file. I.e. don't parse out 'hard' words.
+```
+#### Examples
+```
+python3 parse_dictionary.py -i ./data/dictionary_compact.json -o ./data/
```

Comment:
So looking in the code base, you are including the `dictionary_compact.json` and all of the individual extracted files.

I'd say what would be a better approach would be to have the script write it's files into some folder named like `dist` or `output`, and include that folder and it's contents in `.gitignore`.

After that, define some format for your `dictionary` file (which could even just match your exact `dictionary_compact.json` format) and document that. Then also .gitignore the dictionary file.

After that, the whole "dictionary" component becomes BYOD (bring your own data)

Reply:
Ok good idea to make things more modular.